synapse/config/tls.py

```diff
@@ -73,6 +89,27 @@ def default_config(self, config_dir_path, server_name, **kwargs):
 
         # Don't bind to the https port
         no_tls: False
+
+        # List of allowed TLS fingerprints for this server to publish along
+        # with the signing keys for this server. Other matrix servers that
+        # make HTTPS requests to this server will check that the TLS
+        # certificates returned by this server match one of the fingerprints.
+        #
+        # Synapse automatically adds its the fingerprint of its own certificate
+        # to the list. So if federation traffic is handle directly by synapse
+        # then no modification to the list is required.
+        #
+        # If synapse is run behind a load balancer that handles the TLS then it
+        # will be necessary to add the fingerprints of the certificates used by
+        # the loadbalancers to this list if they are different to the one
+        # synapse is using.
+        #
+        # Homeservers are permitted to cache the list of TLS fingerprints
+        # returned in the key responses. It may be necessary to publish the
+        # fingerprints of a new certificate and wait for the caches on other
+        # servers to expire before deploying it.
```

Comment:
This worries me a bit if we're not explicit about the cache invaliadtion timings